analyzers/src/SonarAnalyzer.CSharp/Rules/EqualityOnFloatingPoint.cs

```diff
@@ -101,33 +110,29 @@ private static void CheckEquality(SonarSyntaxNodeReportingContext context)
                 && type is INamedTypeSymbol { TypeArguments: { } typeArguments }           // Arguments of TSelf, TOther, TResult
                 && typeArguments.Any(IsFloatingPointNumberType))
             || (type is ITypeParameterSymbol { ConstraintTypes: { } constraintTypes }      // constraints of TSelf or of TSelf, TOther, TResult from IEqualityOperators
-                && constraintTypes.Any(constraint => constraint.DerivesOrImplements(KnownType.System_Numerics_IFloatingPointIeee754_TSelf)));
+                && constraintTypes.Any(x => x.DerivesOrImplements(KnownType.System_Numerics_IFloatingPointIeee754_TSelf)));
 
         private static BinaryExpressionSyntax TryGetBinaryExpression(ExpressionSyntax expression) =>
             expression.RemoveParentheses() as BinaryExpressionSyntax;
 
         private static bool IsIndirectInequality(SemanticModel semanticModel, BinaryExpressionSyntax binaryExpression, BinaryExpressionSyntax left, BinaryExpressionSyntax right) =>
             binaryExpression.IsKind(SyntaxKind.LogicalOrExpression)
-                && HasAppropriateOperatorsForInequality(left, right)
-                && HasFloatingType(semanticModel, right.Left, right.Right);
+            && IsOperatorPair(left, right, SyntaxKind.GreaterThanToken, SyntaxKind.LessThanToken)
+            && HasFloatingType(semanticModel, right.Left, right.Right);
 
         private static bool IsIndirectEquality(SemanticModel semanticModel, BinaryExpressionSyntax binaryExpression, BinaryExpressionSyntax left, BinaryExpressionSyntax right) =>
             binaryExpression.IsKind(SyntaxKind.LogicalAndExpression)
-                && HasAppropriateOperatorsForEquality(left, right)
-                && HasFloatingType(semanticModel, right.Left, right.Right);
+            && IsOperatorPair(left, right, SyntaxKind.GreaterThanEqualsToken, SyntaxKind.LessThanEqualsToken)
+            && HasFloatingType(semanticModel, right.Left, right.Right);
 
         private static bool HasFloatingType(SemanticModel semanticModel, ExpressionSyntax left, ExpressionSyntax right) =>
```

Comment:
Both invocations are decomposing binary into left/right anyway
```suggestion
        private static bool HasFloatingType(SemanticModel semanticModel, BinaryExpressionSyntax binary) =>
```